src/Core/Grid/Query/ProductQueryBuilder.php

```diff
@@ -0,0 +1,275 @@
+                'p',
+                $this->dbPrefix . 'product_lang',
+                'pl',
+                'pl.`id_product` = p.`id_product` AND pl.`id_lang` = :id_lang AND pl.`id_shop` = :id_shop'
+            )
+            ->leftJoin(
+                'ps',
+                $this->dbPrefix . 'category_lang',
+                'cl',
+                'cl.`id_category` = ps.`id_category_default` AND cl.`id_lang` = :id_lang AND cl.`id_shop` = :id_shop'
+            )
+            ->leftJoin(
+                'ps',
+                $this->dbPrefix . 'image_shop',
+                'img_shop',
+                'img_shop.`id_product` = ps.`id_product` AND img_shop.`cover` = 1 AND img_shop.`id_shop` = :id_shop'
+            )
+            ->andWhere('p.`state`=1')
+        ;
+
+        $isStockManagementEnabled = $this->configuration->getBoolean('PS_STOCK_MANAGEMENT');
+
+        if ($isStockManagementEnabled) {
+            $stockOnCondition =
+                'sa.`id_product` = p.`id_product`
+                    AND sa.`id_product_attribute` = 0
+                ';
+
+            if ($this->isStockSharingBetweenShopGroupEnabled) {
+                $stockOnCondition .= '
+                     AND sa.`id_shop` = 0 AND sa.`id_shop_group` = :id_shop_group
+                ';
+            } else {
+                $stockOnCondition .= '
+                     AND sa.`id_shop` = :id_shop AND sa.`id_shop_group` = 0
+                ';
+            }
+
+            $qb->leftJoin(
+                'p',
+                $this->dbPrefix . 'stock_available',
+                'sa',
+                $stockOnCondition
+            );
+
+            $qb->setParameter('id_shop_group', $this->contextShopGroupId);
+        }
+
+        $sqlFilters = new SqlFilters();
+        $sqlFilters
+            ->addFilter(
+                'id_product',
+                'p.`id_product`',
+                SqlFilters::MIN_MAX
+            )
+            ->addFilter(
+                'price_tax_excluded',
+                'ps.`price`',
+                SqlFilters::MIN_MAX
+            )
+        ;
+
+        if ($isStockManagementEnabled) {
+            $sqlFilters
+                ->addFilter(
+                    'quantity',
+                    'sa.`quantity`',
+                    SqlFilters::MIN_MAX
+                )
+            ;
+        }
+
+        $this->filterApplicator->apply($qb, $sqlFilters, $filterValues);
+
+        $qb->setParameter('id_shop', $this->contextShopId);
+        $qb->setParameter('id_lang', $this->contextLanguageId);
+
+        foreach ($filterValues as $filterName => $filter) {
+            if (!in_array($filterName, $availableFilters, true)) {
+                continue;
+            }
+
+            if ('active' === $filterName) {
+                $qb->andWhere('ps.`active` = :active');
+                $qb->setParameter('active', $filter);
+
+                continue;
+            }
+
+            if ('name' === $filterName) {
```

Comment:
But since you filter the unknown fields, and `active` is the only one with strict equality I guess you could factorize those three ones with a single one with something like:
```suggestion
            $qb->andWhere(sprintf('pl.`%s` LIKE :%s', $filterName));
            $qb->setParameter($filterName, '%' . $filter . '%');
```

But both work so I don't mind what you choose in this case

Reply:
With only 3 conditions I would like to keep them separate, it's not that much code and IMO it reads a little bit easier.

Reply:
Ok, no problem!